syntax = "proto3";
package whispernet;

enum MessageType {
    CONNECTION_REQUEST = 0;
    CONNECTION_RESPONSE = 1;
    CHAT_MESSAGE = 2;
}

message WhisperMessage {
    MessageType type = 1;
    oneof payload {
        ConnectionRequest connection_request = 2;
        ConnectionResponse connection_response = 3;
        ChatMessage chat_message = 4;
    }
}

message ConnectionRequest {
    string username = 1;   
    string did = 2;       
}

  message ConnectionResponse {
    bool accepted = 1;
    string message = 2; 
  }
  
  message ChatMessage {
    string sender = 1;
    string content = 2;
  }


